service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/NoticeOfProceedingsControllerAboutToSubmitTest.java

```diff
@@ -0,0 +1,117 @@
+@OverrideAutoConfiguration(enabled = true)
+class NoticeOfProceedingsControllerAboutToSubmitTest {
+
+    private static final String AUTH_TOKEN = "Bearer token";
+    private static final String USER_ID = "1";
+
+    @MockBean
+    private CaseDataExtractionService caseDataExtractionService;
+
+    @MockBean
+    private DocmosisDocumentGeneratorService docmosisDocumentGeneratorService;
+
+    @MockBean
+    private UploadDocumentService uploadDocumentService;
+
+    @Autowired
+    private ObjectMapper mapper;
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @Test
+    void shouldGenerateC6NoticeOfProceedingsDocument() throws Exception {
+        byte[] pdf = {1, 2, 3, 4, 5};
+        Document document = c6Document();
+        DocmosisDocument docmosisDocument = DocmosisDocument.builder()
+            .bytes(pdf)
+            .documentTitle("notice_of_proceedings_(C6)")
+            .build();
+
+        Map<String, String> templateData = createTemplatePlaceholders();
+
+        given(caseDataExtractionService.getNoticeOfProceedingTemplateData(any(), any()))
+            .willReturn(templateData);
+        given(docmosisDocumentGeneratorService.generateDocmosisDocument(templateData, C6))
+            .willReturn(docmosisDocument);
+        given(uploadDocumentService.uploadPDF(USER_ID, AUTH_TOKEN, pdf, "notice_of_proceedings_(C6).pdf"))
+            .willReturn(document);
+
+        MvcResult response = makeRequest();
+
+        AboutToStartOrSubmitCallbackResponse callbackResponse = mapper.readValue(response.getResponse()
+            .getContentAsByteArray(), AboutToStartOrSubmitCallbackResponse.class);
+
+        CaseData caseData = mapper.convertValue(callbackResponse.getData(), CaseData.class);
+
+        DocumentReference noticeOfProceedingBundle = caseData.getNoticeOfProceedingsBundle().get(0).getValue()
+            .getDocument();
+
+        assertThat(noticeOfProceedingBundle.getDocument_url()).isEqualTo(document.links.self.href);
+        assertThat(noticeOfProceedingBundle.getDocument_filename()).isEqualTo(document.originalDocumentName);
+        assertThat(noticeOfProceedingBundle.getDocument_binary_url()).isEqualTo(document.links.binary.href);
+    }
+
+    private MvcResult makeRequest() throws Exception {
+        return mockMvc
+            .perform(post("/callback/notice-of-proceedings/about-to-submit")
+                .header("authorization", AUTH_TOKEN)
+                .header("user-id", USER_ID)
+                .contentType(MediaType.APPLICATION_JSON)
+                .content(readBytes("fixtures/C6CaseData.json")))
+            .andExpect(status().isOk())
+            .andReturn();
+    }
+
+    private Map<String, String> createTemplatePlaceholders() {
+        return Map.of(
+            "jurisdiction", "PUBLICLAW",
+            "familyManCaseNumber", "SW123123",
+            "applicantName", "James Nelson",
+            "orderTypes", "Care order",
+            "childrenNames", "James Nelson",
```

Comment:
I would request this be changed to either suits characters or characters from Sherlock Holmes.